### DISCUSSION TITLE
need suggestion on database and tool sets of MLOPTs

### DISCUSSION THREAD

**Discussion by enthusiastics** [vaexio/vaex#1554] [General]
Hi, 

I just found Vaex and I think this is an extremely valuable technology, especially for my use case where we require very fast and scalable computation. I actually want to put Vaex into production. But I would like to ask what is the most compatible streaming data tools and database to use (basically, I would need to stream data, apply realtime transformation, output real time prediction, and also store results into database for future reference (or future restarts.))? I really want to apply Vaex to tick data where there is massive volume and per second and high speed analysis is needed.

**Comment by yohplala** [comment]
Hello @enthusiastics ,

_Disclaimer: author of [oups](https://github.com/yohplala/oups) and contributor to [fastparquet](https://github.com/dask/fastparquet) here, but not a great packager, and only coding in spare time :)_

As you request _suggestion_, I would _suggest_ [oups](https://github.com/yohplala/oups). It might not be the ideal answer, and I have not battle-proved it yet for this purpose, but I have such application in mind when implementing it.

- it accepts and delivers pandas and/or vaex dataframes,
- it uses parquet format, hive file scheme to store data in chunk (one file is one chunk),
- it uses fastparquet to write parquet files (which I could assess as faster than vaex)
- there is yet potential in fastparquet for faster writings, but I don't plan to tackle that in the short term
- for reading in vaex format, this is pure vaex, and I have not delved in how vaex is handling the topic, nor do I plan to do it in the short term
- it has an `update` mode for coalescing data in larger chunks or to insert/update/correct data within (begining/middle/end, wherever the data is located) of an ordered dataset
- it cannot remove rows however (to be done outside of `oups` with `fastparquet` for instance), only replace existing ones or add new ones.
- depending others user-submitted requests / use cases / examples, it could be extended if necessary/relevant.

All this to say if in terms of speed, it does not match your need, there might be yet some potential, but not for the short term. My next move is to integrate it as backbone in my own data processing framework (based on numpy, pandas, vaex...) now that I have reached the MPV in terms of features.
